### ISSUE TITLE
Mixed issues not being reported

### ISSUE THREAD

**Issue by ADmad** [vimeo/psalm#9619]
For the following config

```xml
<?xml version="1.0"?>
<psalm
    errorLevel="5"
    resolveFromConfigFile="true"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns="https://getpsalm.org/schema/config"
    xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
    findUnusedBaselineEntry="false"
    findUnusedCode="false"
    usePhpDocPropertiesWithoutMagicCall="true"
    reportMixedIssues="true"
>
    <projectFiles>
        <directory name="src" />
        <ignoreFiles>
            <directory name="vendor" />
        </ignoreFiles>
    </projectFiles>
    <issueHandlers>
        <MixedArgument errorLevel="error" />
        <MixedAssignment errorLevel="error" />
        <MixedPropertyAssignment errorLevel="error" />
    </issueHandlers>
</psalm>
```
running psalm for this code 

```php
<?php

/**
 * Safely access the values in $this->params.
 *
 * @param string $name The name or dotted path to parameter.
 * @param mixed $default The default value if `$name` is not set. Default `null`.
 * @return mixed
 */
function getParam(string $name, $default = null)
{
    if ($name && $default) {} // suppress unused name warnings
    return null;
}

$type = getParam('type');
$type = strtolower($type);

echo $type;
```
I would expect the same errors to be reported as seen on https://psalm.dev/r/34a696dcc4

yet I get ` No errors found!`.

**Comment by orklah** [comment]
Can you try to see if you get them at `errorLevel="1"`?

**Comment by ADmad** [comment]
Yes I do the same results as the playground with `errorLevel="1"` but that's not want I want.

I expect using `reportMixedIssues="true"` (and explicitly enabling the specific handlers) to show the mixed errors as documented here https://psalm.dev/docs/running_psalm/configuration/#reportmixedissues.

**Comment by orklah** [comment]
Yeah, seems like the doc is wrongly worded. It seems currently that it was only made for silencing mixed issues. For legacy reasons, it's somewhat of a mess.

Note: I don't know what your motives are for doing that, it seems like a bad idea to me on first glance. A lot of issues above level 5 will cause mixed to appear out of nowhere. If you don't display those, you'll see the mixed part without seeing the cause of it.

However, if you want to fix things, it should not be too difficult. The main thing to understand is: https://github.com/vimeo/psalm/blob/7b541ed74f40f06e84a56dde0eb4bee34eef6e8d/src/Psalm/Config.php#L1082 
This is what turns a config in xml into a property of Config. Once you get that, the property is used twice. It should be easy to track that and find out why it doesn't behave properly

**Comment by ADmad** [comment]
> Note: I don't know what your motives are for doing that, it seems like a bad idea to me on first glance. A lot of issues above level 5 will cause mixed to appear out of nowhere. If you don't display those, you'll see the mixed part without seeing the cause of it.

I am working with an existing codebase and even level 4 generates hundreds of errors and going through the list is too tedious. Passing mixed type values to arguments which expects string (like the various string functions) is a common type of error that's occurring after turning on strict typing, so I was hoping to turn on just the related issue handlers and potentially get a more manageable list of errors.

> However, if you want to fix things, it should not be too difficult.

I'll see if I can find the time to dig into it :slightly_smiling_face:.

**Comment by orklah** [comment]
> I am working with an existing codebase and even level 4 generates hundreds of errors

Seems like a job for the baseline :). Just ignore everything and fix incoming errors as they arrive

In my experience, on existing codebase, the list of Mixed issue is rarely the shortest!

**Comment by ADmad** [comment]
> Just ignore everything...

The problem is we have (naively) added `declare(strict_types=1)` too already too all files, which is causing multiple `TypeError`s which need to be addressed and can't be brushed under the rug :smile:.

**Comment by orklah** [comment]
You might want to take a look at https://github.com/orklah/psalm-strict-types

This is very much not perfect, but it should tell you if you have flagrant issues with strict_types and also tell you if entire files are safe to add strict_types.

**Comment by ADmad** [comment]
> You might want to take a look at https://github.com/orklah/psalm-strict-types

Thank you!! I'll give it a shot.

**Comment by ADmad** [comment]
The plugin has `"vimeo/psalm": "^4.4.0"`.

So can't try it with psalm 5.9 :disappointed:

**Comment by orklah** [comment]
I think it would be ready to go for Psalm 5, all is missing is someone to test it so if you're up for it, just try it!
